File: app/src/main/java/com/expensio/domain/model/Group.kt
package com.expensio.domain.model

data class Group(
    val id: String,
    val name: String,
    val createdBy: String?,
    val createdAt: String,
    val memberCount: Int,
)

data class GroupMember(
    val id: String,
    val name: String,
    val isGuest: Boolean,
    val userId: String?,
    val guestId: String?,
    val profilePic: String?,
)

data class GroupDetail(
    val id: String,
    val name: String,
    val createdBy: String?,
    val createdAt: String,
    val members: List<GroupMember>,
)

data class UserSearchResult(
    val id: String,
    val name: String,
    val email: String?,
    val profilePic: String?,
)

File: backend/app/schemas/group.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class GroupCreate(BaseModel):
    name: str


class MemberResponse(BaseModel):
    id: uuid.UUID
    name: str
    is_guest: bool
    user_id: Optional[uuid.UUID] = None
    guest_id: Optional[uuid.UUID] = None
    profile_pic: Optional[str] = None

    model_config = {"from_attributes": True}


class GroupResponse(BaseModel):
    id: uuid.UUID
    name: str
    created_by: Optional[uuid.UUID]
    created_at: datetime
    member_count: int

    model_config = {"from_attributes": True}


class GroupDetailResponse(BaseModel):
    id: uuid.UUID
    name: str
    created_by: Optional[uuid.UUID]
    created_at: datetime
    members: list[MemberResponse]

    model_config = {"from_attributes": True}


class AddMemberRequest(BaseModel):
    email: Optional[str] = None
    guest_name: Optional[str] = None
